src/Impostor.Server/Net/ClientBase.cs

```diff
@@ -1,29 +1,49 @@
 ﻿using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
+using Impostor.Api;
 using Impostor.Api.Innersloth;
 using Impostor.Api.Net;
 using Impostor.Api.Net.Messages;
 using Impostor.Api.Net.Messages.S2C;
+using Impostor.Api.Reactor;
 using Impostor.Hazel;
 using Impostor.Server.Net.State;
 
 namespace Impostor.Server.Net
 {
     internal abstract class ClientBase : IClient
     {
-        protected ClientBase(string name, IHazelConnection connection)
+        protected ClientBase(string name, IHazelConnection connection, ISet<Mod> mods)
         {
             Name = name;
             Connection = connection;
+            Mods = mods;
             Items = new ConcurrentDictionary<object, object>();
+
+            ModIdMap = new Dictionary<int, string>();
+
+            var i = -1;
+
+            foreach (var mod in mods.OrderBy(x => x.Id))
```

Comment:
This looks confusing as hell, maybe make a method about whatever this is trying to do

Reply:
Mod id mapping will be remade anyways